packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -332,7 +419,25 @@ export class ChainService implements ChainServiceInterface {
             to: tokenContract.address,
           };
 
-          await this.sendTransaction(approveAllowanceRequest);
+          const tx = await this.sendTransaction(approveAllowanceRequest);
+
+          this.logger.info(
+            {
+              assetHolderAddress,
+              tokenAddress,
+              amount,
+              tx: tx.hash,
+            },
+            'Submitted transaction to approve maximum amount of asset holder spending'
+          );
+
+          await tx.wait();
```

Comment:
Why wait for transaction mining here? The wallet **can** await requests to the chain service. So the expectation is for chain service requests to return in a reasonable amount of time. At the moment, the heuristic is that it is reasonable to wait for transaction submission, but not transaction mining.

Reply:
I think this is just a bad merge conflict resolution; it used to call `.wait()` before the recent refactor